app/models/concerns/slugifiable.rb

```diff
@@ -0,0 +1,13 @@
+module Slugifiable
+  module InstanceMethods
+    def slug
+      username.downcase.gsub(" ","-")
+    end
+  end
+
+  module ClassMethods
+    def find_by_slug(slug)
+      all.find { |instance| instance.slug == slug }
```

Comment:
Instead of redefining your own `find_by_slug` method, you can utilize the existing ActiveRecord `find_by` method.

```ruby
def find_by_slug(slug)
  find_by(slug: slug)
end
```
This returns an ActiveRecord::Collection object, allowing you to chain additional ActiveRecord methods.